staging/src/k8s.io/apimachinery/pkg/util/proxy/transport_test.go

```diff
@@ -274,3 +275,34 @@ func TestProxyTransport(t *testing.T) {
 		testItem(name, &item)
 	}
 }
+
+func TestRewriteResponse(t *testing.T) {
+	testTransport := &Transport{
+		Scheme:      "http",
+		Host:        "foo.com",
+		PathPrepend: "/proxy/node/node1:10250",
+	}
+	encodeType := []string{"gzip", "deflate"}
+	resp := &http.Response{
+		Body: ioutil.NopCloser(bytes.NewBufferString("Unit test")),
```

Comment:
the body has to actually be gzip or deflate-encoded for  the rewriters to work properly